### DISCUSSION TITLE
Features that prisma mongodb does not support yet because of driver limitation

### DISCUSSION THREAD

**Discussion by masterbater** [prisma/prisma#12746] [Q&A]
Can you guys please list some of features that is not yet supported because of Rust driver
Correct me if im wrong but this are some features I know that is not yet supported by rust driver
- Client Side Field Level Encryption CSFLE (I check this is their priority in jira)
- Change streams?

**Comment by matthewmueller** [comment]
Thanks, we should indeed start compiling a list. The one top of mind for us is:
- [Decimal support](https://github.com/prisma/prisma/issues/12637)

You might be able to use the [raw commands](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#raw-queries-with-mongodb) to support these features.

They use a low-level driver API and may just pass through the Rust Driver directly to the database. If you give it a try, please let us know how it fares for you!

**Comment by khowling** [comment]
Would like to use mongo Change streams with Prisma, or at least some way to get the underlying mongodb connection from Prisma so I can do my own `db.watch()`

**Reply by gzsig** [reply]
hey @khowling I am having the same problem, I need to be able to watch a specific mongo collection and my project uses prisma. Did you ever get around to finding a solution for this? Thanks :)

**Reply by khowling** [reply]
Hi, no sorry, I didn't.  I've actually moved away from using Prisma now, it wasn't adding a ton of value to by project with mongo, and I needed to get hold of the connection.

**Reply by gregg-cbs** [reply]
Damn I need change streams with `db.watch` and i realize if create another mongodb driver instance just to listen to changes the data wont be exactly the same as what comes through prisma. for example `_id` and `id` field. Relations etc.

Any feedback from the Prisma team on whats best to do in this scenario?

**Comment by masterbater** [comment]
If you use nodejs driver when you received the stream just change the_id to
id field. map fields wont work since the raw document will always be feed
on streams. Until prisma client adds change streams support you can do that
simple workaround. FYI prisma uses rust driver not node

**Reply by gregg-cbs** [reply]
Thanks, thats what I tried to do until Nextjs doesnt have a server start hook to setup the watchers... so im screwed either way.

**Comment by momorsi55573** [comment]
fuck

**Comment by dev-talha-anwar** [comment]
any update on Change streams?

**Comment by habasefa** [comment]
Is there any update on change streams?

**Comment by ItzSiL3Nce** [comment]
- Updating and deleting records without bothering/altering their relations
(this is MongoDB's default but the driver augments this functionality without having a way to disable it and it HITS performance in write operations **A LOT**)

I've already made a pull request for this over a year ago:
https://github.com/prisma/prisma-engines/pull/4116

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
